File: Pancake Sort/PancakeSort.py
def flip(arr, i):
    start = 0

    while start < i:
        arr[start], arr[i] = arr[i], arr[start]
        start += 1
        i -= 1

def find_max(arr, n):
    max_index = 0

    for i in range(1, n):
        if arr[i] > arr[max_index]:
            max_index = i

    return max_index

def pancake_sort(arr):

    n = len(arr)

    for curr_size in range(n, 1, -1):

        max_index = find_max(arr, curr_size)

        if max_index != curr_size - 1:

            flip(arr, max_index)

            flip(arr, curr_size - 1)

arr = [23, 10, 20, 11, 12, 6, 7]

pancake_sort(arr)

print(arr)

File: Pancake Sort/PancakeSort.java
class PancakeSort {

    static void flip(int arr[], int i) {
        int start = 0;

        while(start < i) {

            int temp = arr[start];
            arr[start] = arr[i];
            arr[i] = temp;

            start++;
            i--;
        }
    }

    static int findMax(int arr[], int n) {
        int maxIndex = 0;

        for(int i = 1; i < n; i++) {
            if(arr[i] > arr[maxIndex])
                maxIndex = i;
        }

        return maxIndex;
    }

    static void pancakeSort(int arr[], int n) {

        for(int currSize = n; currSize > 1; currSize--) {

            int maxIndex = findMax(arr, currSize);

            if(maxIndex != currSize - 1) {

                flip(arr, maxIndex);

                flip(arr, currSize - 1);
            }
        }
    }

    public static void main(String[] args) {

        int arr[] = {23, 10, 20, 11, 12, 6, 7};

        pancakeSort(arr, arr.length);

        for(int num : arr)
            System.out.print(num + " ");
    }
}
